### DISCUSSION TITLE
IPv6 Support

### DISCUSSION THREAD

**Discussion by st1905** [pgj/freebsd-wifibox#62] [Q&A] [answered]
When i login to console, i can see that the global ipv6 address is assigned and its working but on FreeBSD, v6 address is not available, is there something i`m missing ?

**Comment by pgj** [comment]
Hi there: The IPv6 support is not configured and it is not tested officially.  I think it is doable but it has not been in the scope.  I can create a ticket about it and track the related work there.

**Reply by pgj** [reply]
I have created #63 for this.

**Comment by pgj** [comment]
I tried to explore this on my own configuration, where IPv6 is supported in theory.  But does `ping6` work for you from the Wifibox console?  For me, it does not -- though it finds the IPv6 address, no responses are registered.  When `ping6`ing an internal host, it works.

**Comment by st1905** [comment]
Hi, Yes it works perfectly, no packet loss or anything like that. ping6 works inside wifibox console.

**Reply by pgj** [reply]
I had to replace `udhcpd` for `dhcpcd` and enable SLAAC on the router and now I can do a `ping6 google.com` without problems.  Without SLAAC, manual assignment of the IPv6 address also worked, basically out of the box.  What I could not solve is the gateway configuration -- the FreeBSD host cannot get either a global IPv6 address or have any traffic forwarded.

**Reply by st1905** [reply]
Is there a possibility that its working here because i have static IPv6 route/gateway configured on the openwrt router and freebsd picks that up as the v6 gateway ?

**Reply by pgj** [reply]
Sorry, what is working?  I thought your original issue is that same as mine: you do not have an IPv6 address on the FreeBSD host and there is no such traffic.

**Reply by st1905** [reply]
That`s correct, on my laptop where wifibox is installed there is no ipv6 address on FreeBSD only in wifibox console/vm has the address assigned and it works fine.

on my pc where there is no wifibox, the FreeBSD network interface has the ipv6 address, that`s what i meant, sorry for the confusion.

**Comment by st1905** [comment]
My IPv6 provider is not my ISP, i use a wireguard connection to distribute global IPv6 addresses to the openwrt clients, SLAAC being used.

This is my current setup if it helps https://otus.aug.is/route48-openwrt-wireguard  (Route 48 is no longer active(its closed due to abuse) so i configured my own vps to do the same which is working quite good).

Not sure if this helps.

**Comment by pgj** [comment]
I was able to make this work, see my related comment in #63.  I will soon publish a version of `wifibox-alpine` that you can test.

**Reply by st1905** [reply]
I`m wondering if it will ever be possible to enable bridged interfaces between host/guest instead of nat ?

**Reply by pgj** [reply]
I have investigated this topic for the IPv4 case and I could not find a satisfying solution.  If I recall correctly, the main problem is that the `eth0` and `wlan0` interfaces cannot be bridged because [_"most most Access Points (APs) will reject frames that have a source address that didn’t authenticate with the AP. Since Linux does ethernet bridging transparently (doesn’t modify outgoing or incoming frames)"_](https://wiki.debian.org/BridgeNetworkConnections#Bridging_with_a_wireless_NIC).  In theory, `ebtables` could help with this but I could not make it work.

**Comment by pgj** [comment, marked as answer]
Since there has been added support for IPv6, I am now considering this question answered.  I accept that there might be other ways to implement this functionality but I would put that in a scope of a different exploration.

**Reply by st1905** [reply]
i used the latest wifibox-alpine image and it worked, sometimes when i restart, it loses the IPV6 connectivity. That`s the only issue i have, when that happens, i have to stop network, restart wifibox and start network again.
